tests/devices/test_default_qubit.py

```diff
@@ -870,6 +871,337 @@ def circuit():
+                            0.0 + 0.0j,
+                            -1.11700225 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                        ],
+                        [
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            -0.44058791 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                        ],
+                        [
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            -0.44058791 + 0.0j,
+                            0.0 + 0.0j,
+                        ],
+                        [
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.0 + 0.0j,
+                            0.93441396 + 0.0j,
+                        ],
+                    ]
+                ),
+                -1.1373060481,
+            ),
+        ],
+    )
+    def test_sparse_hamiltonian_expval(self, qubits, operations, hamiltonian, expected_output):
+        """Test that expectation values of sparse hamiltonians are properly calculated."""
+
+        hamiltonian = scipy.sparse.coo_matrix(hamiltonian)
+
+        dev = qml.device("default.qubit", wires=qubits, shots=None)
+        dev.apply(operations)
+        expval = dev.expval(qml.SparseHamiltonian(hamiltonian, wires=[0]))[0]
+
+        assert (expval, expected_output)
```

Comment:
might be good to use `np.allclose` instead, since the assertion will fail even if they differ by a small amount, right?